### ISSUE TITLE
some question

### ISSUE THREAD

**Issue by Fu-Zhou-2019** [YingYuChen97/CSCA#4]
```
 with torch.no_grad():

            for i in range(opt.way):
                sim_matrix = [[] for _ in range(opt.way)]
                
                for j in range(opt.way):
                    output = model[i](mtest_data).squeeze(-1).squeeze(-1)
                    sim_matrix[j] = cos(output[i * opt.query:(i+1) * opt.query],prototypes[j]).cpu().data.numpy()
                
                pred = np.argmax(sim_matrix,axis=0)
                correct += np.count_nonzero(pred==i)
```

In this code, you seem to have implied the query's label. When the outer loop i is fixed, the inner loop send the query samples belonging to class i to model[i]. 
What about when a query sample that does not belong to class i is sended in the model[i]?

_Originally posted by @Fu-Zhou-2019 in https://github.com/YingYuChen97/CSCA/issues/2#issuecomment-1248083645_

**Comment by akito047** [comment]
Isn't their model[i] learned on the class `i` samples in the support set? If that is the case, then this part of the code is fine I think. You should have the same class samples in the query and the support set.

**Comment by Fu-Zhou-2019** [comment]
Yes, as you said, we have the same class samples in the query and support set, but you don't know the real class of the query when you predict it.
I ran their code and their query format is [class1*15, class2*15,···, class5*15]， for a total of seventy-five query samples.
In the code, for the outer loop fixed to i, they just take out the fifteen query samples which belong to class i and send them to model[i], then compute the similarity using prototype[1],prototype[2]···prototype[way]. This is incorrect because you don't know that these queries are belonging to class i when you predict them. You should send the queries to model[1], model[2]····model[way] and calculate the similarity with prototype[1],prototype[2]····prototype[way] respectively.
The code should be:
```
for j in range(opt.way):
                    output = model[j](mtest_data).squeeze(-1).squeeze(-1)
                    sim_matrix[j] = cos(output[i * opt.query:(i+1) * opt.query],prototypes[j]).cpu().data.numpy()
```
  not :
```
for j in range(opt.way):
                    output = model[i](mtest_data).squeeze(-1).squeeze(-1)
                    sim_matrix[j] = cos(output[i * opt.query:(i+1) * opt.query],prototypes[j]).cpu().data.numpy()
```

**Comment by YingYuChen97** [comment]
@Fu-Zhou-2019 
It seems I made a mistake here. I'll close this repo first until I solve it.
Thanks for replying.
